src/server/v2.0/handler/auditlog.go

```diff
@@ -111,14 +115,89 @@ func (a *auditlogAPI) ListAuditLogs(ctx context.Context, params auditlog.ListAud
 		WithPayload(auditLogs)
 }
 func (a *auditlogAPI) ListAuditLogExts(ctx context.Context, params auditlog.ListAuditLogExtsParams) middleware.Responder {
-	// TODO: implement this method
+	secCtx, ok := security.FromContext(ctx)
+	if !ok {
+		return a.SendError(ctx, errors.UnauthorizedError(errors.New("security context not found")))
+	}
+	if !secCtx.IsAuthenticated() {
+		return a.SendError(ctx, errors.UnauthorizedError(nil).WithMessage(secCtx.GetUsername()))
+	}
+	query, err := a.BuildQuery(ctx, params.Q, params.Sort, params.Page, params.PageSize)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	if err := a.RequireSystemAccess(ctx, rbac.ActionList, rbac.ResourceAuditLog); err != nil {
+		ol := &q.OrList{}
+		if sc, ok := secCtx.(*local.SecurityContext); ok && sc.IsAuthenticated() {
+			user := sc.User()
+			member := &project.MemberQuery{
+				UserID:   user.UserID,
+				GroupIDs: user.GroupIDs,
+			}
+
+			projects, err := a.projectCtl.List(ctx, q.New(q.KeyWords{"member": member}), project.Metadata(false))
+			if err != nil {
+				return a.SendError(ctx, fmt.Errorf(
+					"failed to get projects of user %s: %v", secCtx.GetUsername(), err))
+			}
+			for _, project := range projects {
+				if a.HasProjectPermission(ctx, project.ProjectID, rbac.ActionList, rbac.ResourceLog) {
```

Comment:
if a user has the permission of the overall system, why do we still need to validate the permission of projects.